quadriga/src/test/java/edu/asu/diging/quadriga/core/service/impl/SimpleUserAppServiceImplTest.java

```diff
@@ -0,0 +1,157 @@
+package edu.asu.diging.quadriga.core.service.impl;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+
+import edu.asu.diging.quadriga.core.citesphere.CitesphereConnector;
+import edu.asu.diging.quadriga.core.data.SimpleUserAppRepository;
+import edu.asu.diging.quadriga.core.model.citesphere.CitesphereAppInfo;
+import edu.asu.diging.quadriga.core.model.users.SimpleUserApp;
+import edu.asu.diging.simpleusers.core.model.impl.SimpleUser;
+
+public class SimpleUserAppServiceImplTest {
+
+    @InjectMocks
+    private SimpleUserAppServiceImpl simpleUserAppService;
+
+    @Mock
+    private CitesphereConnector citesphereConnector;
+
+    @Mock
+    private SimpleUserAppRepository simpleUserAppRepository;
+
+    private String USER_1;
+    private String USER_2;
+    private SimpleUser SIMPLE_USER_1;
+    private String CLIENT_ID_1;
+    private String CLIENT_ID_2;
+    private String CLIENT_ID_3;
+    private SimpleUserApp USER_APP_1;
+    private SimpleUserApp USER_APP_2;
+    private List<SimpleUserApp> USER_APP_LIST;
+    private CitesphereAppInfo APP_1;
+    private CitesphereAppInfo APP_2;
+    private CitesphereAppInfo APP_3;
+
+    @Before
+    public void setUp() {
+        MockitoAnnotations.openMocks(this);
+
+        USER_1 = "user1";
+        USER_2 = "user2";
+
+        SIMPLE_USER_1 = new SimpleUser();
+        SIMPLE_USER_1.setUsername(USER_1);
+
+        CLIENT_ID_1 = "clientId1";
+        CLIENT_ID_2 = "clientId2";
+        CLIENT_ID_3 = "clientId3";
+        
+        USER_APP_LIST = new ArrayList<>();
+
+        USER_APP_1 = new SimpleUserApp();
+        USER_APP_1.setId("1");
+        USER_APP_1.setUsername(USER_1);
+        USER_APP_1.setAppClientId(CLIENT_ID_1);
+        USER_APP_LIST.add(USER_APP_1);
+
+        USER_APP_2 = new SimpleUserApp();
+        USER_APP_2.setId("2");
+        USER_APP_2.setUsername(USER_1);
+        USER_APP_2.setAppClientId(CLIENT_ID_2);
+        USER_APP_LIST.add(USER_APP_2);
+
+        APP_1 = new CitesphereAppInfo();
+        APP_1.setClientId(CLIENT_ID_1);
+
+        APP_2 = new CitesphereAppInfo();
+        APP_2.setClientId(CLIENT_ID_2);
+
+        APP_3 = new CitesphereAppInfo();
+        APP_3.setClientId(CLIENT_ID_3);
+    }
+
+    @Test
+    public void test_findByUsername_success() {
+        Mockito.when(simpleUserAppRepository.findByUsername(USER_1)).thenReturn(USER_APP_LIST);
+        List<SimpleUserApp> response = simpleUserAppService.findByUsername(USER_1);
+        Assert.assertNotNull(response);
+        Assert.assertNotEquals(0, response.size());
```

Comment:
one assert statement that tests that the response size is the correct one (what is it, 2?) is enough.

Reply:
should test that the number of results is correct